modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConnectionProvider.java

```diff
@@ -355,4 +360,44 @@ public ConnectionManagementType getConnectionManagementType() {
     return oauthConfig.getDelegateConnectionProviderModel().getConnectionManagementType();
   }
 
+  @Override
+  public void onBorrow(C connection) {
+    delegatePoolingListener.onBorrow(connection);
+  }
+
+  @Override
+  public void onReturn(C connection) {
+    delegatePoolingListener.onReturn(connection);
+  }
+
+  private PoolingListener<C> getDelegatePoolingListener() {
+    if (unwrappedDelegate instanceof PoolingListener) {
+      return new PoolingListener<C>() {
+
+        @Override
+        public void onBorrow(C connection) {
+          ((PoolingListener) unwrappedDelegate).onBorrow(connection);
+        }
+
+        @Override
+        public void onReturn(C connection) {
+          ((PoolingListener) unwrappedDelegate).onReturn(connection);
+        }
+      };
+    } else {
+      return new PoolingListener<C>() {
+
+        @Override
+        public void onBorrow(C connection) {
+          throw new UnsupportedOperationException("The delegate connection provider does not support Pooling");
```

Comment:
This breaks liskov principle. You should use the NullObject pattern here instead